zoltar-xgboost/src/main/java/ml/dmlc/xgboost4j/java/GompLoader.java

```diff
@@ -43,28 +42,23 @@ private static void loadGomp() throws IOException {
   }
 
   private static void load(final String name) throws IOException {
-    try {
-      final Vector<String> libraries = getLoadedLibraries(ClassLoader.getSystemClassLoader());
-      System.out.println("Loaded libs" + libraries);
-      final String[] libs = libraries.toArray(new String[libraries.size()]);
-
-      final String libFile = name.substring(name.lastIndexOf('/') + 1, name.length());
-      final String libName = libFile.substring(0, libFile.indexOf('.'));
-      if (Arrays.stream(libs).noneMatch(n -> n.contains(libName))) {
-        System.load(NativeLibLoader.createTempFileFromResource(name));
-      }
-    } catch (final IOException e) {
-      throw new IOException(e);
+    final Collection<String> libraries = getLoadedLibraries(ClassLoader.getSystemClassLoader());
+    System.out.println("Loaded libs" + libraries);
+    final String libFile = name.substring(name.lastIndexOf('/') + 1);
+    final String libName = libFile.substring(0, libFile.indexOf('.'));
+    if (libraries.stream().noneMatch(n -> n.contains(libName))) {
+      System.load(NativeLibLoader.createTempFileFromResource(name));
     }
-
   }
 
-  private static Vector<String> getLoadedLibraries(final ClassLoader loader) throws IOException {
+  private static Collection<String> getLoadedLibraries(final ClassLoader loader)
+      throws IOException {
     try {
       final Field libField = ClassLoader.class.getDeclaredField("loadedLibraryNames");
       libField.setAccessible(true);
-
-      return (Vector<String>) libField.get(loader);
+      // As of Java 11, the ClassLoader.loadedLibraryNames is a HashSet. In prior versions it is
```

Comment:
🎨 @rculbertson no need to add this comment here ... we can put it as part of the commit message

Reply:
Sure, I can remove that.